src/coreclr/jit/llvmcodegen.cpp

```diff
@@ -1933,6 +1937,13 @@ void Llvm::buildCall(GenTreeCall* call)
     }
 
     Value* callValue = emitCallOrInvoke(llvmFuncCallee, argVec);
+
+    // Restore the saved shadow stack.
+    if (shadowStackToRestoreValue != nullptr)
+    {
+        _builder.CreateStore(shadowStackToRestoreValue, shadowStackTopAddrValue);
+    }
```

Comment:
This was carried over from the IL importer version of the code, where it was added in https://github.com/dotnet/runtimelab/pull/1791.

If my inference is correct, the reason shadow stack top needs to be restored is so that we don't end up doing this:
```
[Native caller] ; i = 0; ManagedCallee();
[ManagedCallee] ; SS = new (...)]
[Stack grows by X]
[NativeCall] ; SSTop = SS + X

  [Native caller] ; i = 1; ManagedCallee();
  [ManagedCallee] ; SS = SS + X]
  [Stack grows by X]
  [NativeCall] ; SSTop = SS + 2X

    [Native caller] ; i = 2; ManagedCallee();
    [ManagedCallee] ; SS = SS + 2X]
    [Stack grows by X]
    [NativeCall] ; SSTop = SS + 3X

; And so on until SSTop grows out of bounds
```
Presumably, we could restore it in the RPI frame instead though, which would be better since RPI is much less common than PI (once we don't have them everywhere with runtime imports/exports, that is).